client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -61,12 +120,9 @@ const NewMailBoxList = (
 	const isTitan = provider === EmailProvider.Titan;
 
 	// Set visibility for desired hidden fields
-	mailboxes.forEach( ( mailbox ) => {
-		hiddenFieldNames.forEach( ( fieldName ) => {
-			mailbox.setFieldIsVisible( fieldName, false );
-			mailbox.setFieldIsRequired( fieldName, false );
-		} );
-	} );
+	setFieldsVisibilities( mailboxes, hiddenFieldNames );
+	// Set initial values
+	setFieldsInitialValues( mailboxes, initialFieldValues );
```

Comment:
I guess this suggestion is not required, but these lines tripped me up a little bit when I was looking at #64746. Could we maybe wrap this in a `useEffect( () => {}, [] )` to clarify that it should only be executed once? (Assuming I'm correct in that assumption)

Reply:
There was a significant delay and lag when I initially wrapped that in `useEffect`. I'll resolve that in the next PR